salt/_states/metalk8s_kubernetes.py

```diff
@@ -1572,3 +1576,205 @@ def node_uncordoned(name, **kwargs):
         the name of the node
     '''
     return _node_unschedulable(name, False, **kwargs)
+
+
+def _extract_custom_resource_data(name, namespace, body, source, **kwargs):
+    if body and source:
+        raise salt.exceptions.InvalidConfigError(
```

Comment:
Is this an `InvalidConfig`? You catch them yourself later, so I guess it doesn't matter much, but I think some native `ValueError` is more adapted to the "bad arguments" situation you have here (and below). To me, the _Salt invalid config_ really is about minion/master configuration, but I might be wrong.

Reply:
I'll replace with a ValueError exception.